smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/StructuredMemberWriter.java

```diff
@@ -158,41 +170,41 @@ private void writeMapFilterSensitiveLog(TypeScriptWriter writer, MemberShape map
-        String keyParam = "key"; // key of the Object.entries() key-value pair
-        String valueParam = "value"; // value of the Object.entries() key-value pair
-
-        // Reducer is common to all shapes.
-        writer.openBlock("Object.entries($L).reduce(($L: any, [$L, $L]: [string, $T]) => ({", "}), {})",
-            mapParam, accParam, keyParam, valueParam, symbolProvider.toSymbol(mapMember), () -> {
-                writer.write("...$L,", accParam);
-                Shape mapMemberTarget = model.expectShape(mapMember.getTarget());
-                writer.openBlock("[$L]: ", ",", keyParam, () -> {
-                    if (mapMemberTarget instanceof StructureShape) {
-                        writeStructureFilterSensitiveLog(writer, mapMemberTarget, valueParam);
-                    } else if (mapMemberTarget instanceof CollectionShape) {
-                        MemberShape collectionMember = ((CollectionShape) mapMemberTarget).getMember();
-                        writeCollectionFilterSensitiveLog(writer, collectionMember, valueParam);
-                    } else if (mapMemberTarget instanceof MapShape) {
-                        MemberShape nestedMapMember = ((MapShape) mapMemberTarget).getValue();
-                        writeMapFilterSensitiveLog(writer, nestedMapMember, valueParam);
-                    } else {
-                        // This path should not reach because of recursive isMemberOverwriteRequired.
-                        throw new CodegenException(String.format(
-                            "MapFilterSensitiveLog attempted for %s while it was not required",
-                            mapMemberTarget.getType()
-                        ));
-                        // For quick-fix in case of high severity issue:
-                        // comment out the exception above and uncomment the line below.
-                        // writer.write("$L", valueParam);
-                    }
+        } else if (model.expectShape(mapMember.getTarget()) instanceof SimpleShape) {
+            writer.write(mapParam);
+        } else {
+            String accParam = "acc"; // accumulator for the reducer
+            String keyParam = "key"; // key of the Object.entries() key-value pair
+            String valueParam = "value"; // value of the Object.entries() key-value pair
 
-                });
-            }
-        );
+            // Reducer is common to all shapes.
+            writer.openBlock("Object.entries($L).reduce(($L: any, [$L, $L]: [string, $T]) => ({", "}), {})",
+                mapParam, accParam, keyParam, valueParam, symbolProvider.toSymbol(mapMember), () -> {
+                    writer.write("...$L,", accParam);
+                    Shape mapMemberTarget = model.expectShape(mapMember.getTarget());
+                    writer.openBlock("[$L]: ", ",", keyParam, () -> {
+                        if (mapMemberTarget.isStructureShape() || mapMemberTarget.isUnionShape()) {
```

Comment:
This looks really similar to the code in `writeMemberFilterSensitiveLog`, is it possible to refactor this chunk of `if/else` to instead use that? I might be missing something, but the final `else` is the only thing that looks different to me here.

Reply:
Done in commit 4d94416102db3175aa7d93cf2285927ccd639a8a